src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerMetricsDiagnosticListener.cs

```diff
@@ -173,4 +161,16 @@ static bool TryFetchRequest(object payload, out HttpRequestMessage request)
             return true;
         }
     }
+
+    public override void OnEventWritten(string name, object payload)
+    {
+        if (name == OnStopEvent)
```

Comment:
I have changed the order of the `if` conditions. For the common case, we don't want to waste cycles doing a string match for the exception event first.